---
title: "Characterizing parasite generalism illuminates patterns of host-parasite associations"
author:
- Park, A.W.
- Farrell, M.J.
- Schmidt, J.P.
- Huang, S.
- Dallas, T.A.
- Pappalardo, P.
- Drake, J.M.
- Stephens, P.R
- Poulin, R.
- Nunn, C.L.
- Davies, T.J.
csl: biology-letters.csl
header-includes:
   - \usepackage{lineno}
output:
  pdf_document:  default
  word_document: default
  html_document: default
bibliography: relgenREFS.bib
---



\linenumbers



```{r, echo=F}
#get library calls out of the way
library(ggplot2)
library(geiger)
library(MASS)
library(ape)
library(picante)
library(gridExtra)
library(sjPlot)
library(sjmisc)
library(reshape2)
library(xtable)

```

## Abstract

The distribution of parasites among mammalian hosts is complex and represents a differential ability or opportunity to explore the host tree of life. Using data on over 1400 parasite species known to each infect between 1 and 81 mammal host species, we characterize the generalism of parasites using standard effect sizes for three metrics describing host relatedness: average pairwise host phylogenetic distance (PD), maximum PD and degree of aggregation of hosts (the tendency for the hosts of a parasite to cluster within the phylogenetic tree). We explore variation in these metrics in terms of parasite taxonomy and transmission mode. Of the multi-host parasites, the majority are generalist in the sense that the average relatedness of their mammalian host species is roughly equivalent to that expected from a random sample of mammals, with a minority associated with host sets that are more related than expected by chance. The degree of host relatedness is affected by both parasite taxonomy and transmission mode; bacteria and arthropod parasites are typically the most generalist, helminths are intermediate and viruses and protozoa are the most specialist on average. However, characterizing variation within and between groups is important; for example, viruses are more specialist than helminths on average, but the virus group has a larger range on the specialism-generalism continuum, such that the virus group contains some of the most generalist parasites. While not as strong as parasite taxonomy, transmission mode also influences the degree of parasite generalism, with closer relatedness of hosts exhibited for parasites that rely on close contact or complex life-cycle transmission involving intermediate hosts, compared to those with environmental or vector-borne transmission. Both the span of a parasite across the host tree and the tendency to infect clusters of related hosts vary among parasites. For example, all bacteria have large spans, equivalent to random host sets. However, they exhibit the least propensity to infect host clusters within this span. Instead, they appear to typically infect relatively unrelated host species.  The taxonomy and transmission modes of parasites with only one known host species are broadly reflective of multi-host parasites, with rarely single-host viruses being a notable exception. Lastly, a host species' evolutionary distinctiveness is a weak predictor of the number of parasite species asscociated with it. Collectively, this macroecological perspective helps to generate a broad picture of how certain types of parasite and transmission mode are differentially linked with the tendency to associate with multiple host species, captured by a set of complementary metrics defining the dimensions of parasite generalism. 





<!-- Main manuscript findings: 1. parasite type and transmission mode are associated with parasite phylogenetic generalism (the phylogenetic diversity of the permissive host set). 2. Considering raw host number as a measure of specialism discards information about phylogenetic clustering of specificity (i.e., some parasites infect many closely related host species, while others are shotgunned across the host phylo tree).  

Is this correct? 

I don't quite understand how the approach is macroecological. This is mentioned in the abstract and in the final paragraph of the introduction, but never really defined as to how this is macroecological. 

Current formatting is for Biol Letters. Is this the target? If so, perhaps sharpen the manuscript to the above key points, and only keep Figures 1 and 6 in the main text (2 fig max at Biol Letters, 4 displays though if Fig 3 can be a table).

 
-->



## Keywords

Parasite, multi-host, generalism, transmission mode, phylogenetic, macroecology

## Introduction

A generalist parasite has a minimal definition: _able to infect at least two host species_. However, for an equivalent number of host species, parasites may vary in their ability to infect distantly related hosts, differentiating their potential to explore the space of host species along with their ability to infect target species, including humans. The underlying reasons for observing relatedness of hosts of a given parasite include biogeography and co-evolutionary history [@Poulin2008], translating to opportunities for infection, quality resources provided by a host, and limited costs in terms of immune defense and disease-induced mortality. Mechanisms that may allow parasites to overcome host species barriers include rapid evolution and antigenic plasticity [@Woolhouse2002], novel spatial translocation of host or parasite [@Farrell2013], and evolution of host tolerance [@Raberg2009]. Consequently, it is challenging to predict the potential for parasites to acquire hosts that are distantly or closely related to their existing hosts, in spite of the utility for understanding the distribution of parasites and predicting future zoonoses.

<!-- Does first sentence need citation? "explore the space of host species" sounds awkward. I imagine you are referring to the balance between generalism and specialism in this sentence? The evolution of host tolerance is interesting, but the sentence appears to say _Parasites can overcome host species barriers by hosts evolving tolerance mechanisms_, but the parasites don't do anything to overcome the host species barrier there. Am I just misunderstanding? -->

Previous research has defined parasite generalism in alternative ways, including satisfying the criteria of infecting humans and animals [@Woolhouse2002] as well as using host taxonomies to characterize the degree of parasite generalism [@Poulin2003; @Pedersen2005]; for example, classifying parasites by their ability to infect hosts only within a species, or genus, or family, etc. Here, we build on methods to generate continuous measures of generalism [@Poulin2011] using a detailed mammalian host phylogeny to define parasite generalism by the patterns of relatedness of the host sets of parasites. The study is unparalled in the number and diversity of host and parasite species considered together. Known associations between hosts and parasites come from the global mammal parasite database (GMPD) [@Nunn2005; @Stephens2017], from which this study focuses on 430 mammal species representing primates ($n=178$), carnivores ($n=111$) and ungulates ($n=109$). The approach allows both comparison among parasites and calibration against a null model [@Gotelli2000]. From each host set, we calculate the standard effect size of mean pairwise PD between host species, a measure that has previously been used successfully to explain parasite community similarity in primates [@Cooper2012]. In addition, we calculate standard effect sizes for both the maximum PD of host sets and a similar metric that captures the tendency for hosts to be clustered in the phylogenetic tree [@Cooper2008]. 

<!-- you have not defined PD. This also seems to be a mix of introduction and methods -->


We determine that average relatedness of hosts varies with parasite type, defined both taxonomically and by transmission mode, as has been suggested in earlier studies based on categorical definitions of parasite generalism [@Woolhouse2002; @Pedersen2005]. The speciose study, which includes over 1500 parasite species, enables generalism to be characterized further. All taxonomic and transmission mode groupings contain parasites that have large spans in the host tree, measured by standard effect size of maximum PD among hosts. Bacteria, along with environmentally-transmitted and complex life-cycle parasites show the least propensity to infect phylogenetically clustered host groups. Single host parasites ($n$ = 653) are almost as common as multi-host parasites ($n$ = 749) and show similar distributions among parasite taxonomic and transmission mode groupings, with the notable exception of viruses, which are more likely to infect multiple host species. We observe that the number of parasite species known to infect a host species is not well explained by host evolutionary distinctiveness. 

Macroecological approaches to analyzing large host-parasite datasets, such as the one presented here, provide a backdrop against which to consider individual case studies, faciltate an appreciation and understanding of robust patterns of host-parasite associations and extend the concept of parasite generalism to include phenomena such as parasites associating with multiple 'pockets' of related hosts. Further, the study lays out parasite group-level differences in generalism, including within-group variation, which helps navigate to sources of variation that may ultimately be explained by environmental, physiological and behavioral heterogeneity [@Ezenwa2004; @Lee2008; @Poulin2012; @Stein2014]. 

<!-- Currently it seems like the thesis paragraph is the second to last, and the paragraph above sounds a bit like discussion. I may just be too rigid in my manuscript structuring though. The text reads great, but it just doesn't follow the formula I'm used to. -->




## Materials and Methods

Parasite records of occurrence with terrestrial mammals were obtained from GMPD [@Nunn2005; @Stephens2017]. These records include latin binomials and taxonomic classification for host and parasite species, and transmission mode for the majority of parasite species (~80%). Transmission modes were assigned in GMPD based on an extensive literature review [@Nunn2005; @Stephens2017], from which we retain the four heierarchically-dominant transmission modes: close contact, complex life-cycle, environmental and vector-borne (other transmission modes are recorded in GMPD, such as soil-transmission, but these belong to one of the four main transmission modes). GMPD data were manipulated in the R programming environment [@RCoreTeam2016] to establish the number and identity of parasite species per host species and the number and identity of host species per parasite species. 

Additionally, the mammal phylogenetic supertree [@Bininda-Emonds2007] was used to obtain the PD between all pairs of hosts. For the host set of each parasite species, the mean pairwise PD between hosts was calculated, as was its standard effect size [@Webb2002]. The latter was obtained using the R package _Picante_ [@Kembel2010], and a null model in which the community data matrix (host-parasite occurrences) was randomized with the independent swap algorithm [@Gotelli2000] maintaining species occurrence frequency and sample species richness. This measure captures the average relatedness of the host species of a given parasite species on a standardized scale for comparison. Two further standard effect size metrics were calculated on each parasite species' host set in the same way: maximum PD of any two hosts in a parasite's set, and the ratio of the mean minimum PD to the maximum PD. The former metric provides a standardized measure of each parasite's span across the host phylogenetic tree, while the latter provides a measure of the tendency for the host species to exhibit an aggreagated or clumped distribution within the span [@Cooper2008]. This is achieved by calculating the PD of each host species to its nearest host, averaging this across all hosts and scaling by the span. The standard effect size calculations return a $z$-score and a $p$-value, where $z$-scores below -1.96 (specialist) and above +1.96 (generalist) are considered signifcantly different from the null expectation, assuming the null model generates a normally-distributed set of scores [@Ulrich2013]. 

Host evolutionary distinctiveness, measured as millions of years of evolutionary change, was established from the mammal supertree data [@Bininda-Emonds2007] using the _evol.distinct_ function in the R package _Picante_ [@Kembel2010], utilizing the _equal splits_ routine in which shared branches are apportioned equally among descendant lineages [@Redding2006]. Each hosts's terminal branch length was also recorded as an alternate measure of evolutionary distinctiveness.  





## Results


```{r, echo=F, warning=F, message=F}
#mammal supertree

treelist<-read.nexus("data/WR_2005_ST.txt")

#best dates
mtree<-treelist[[1]]

#dist matrix
phydist<-cophenetic(mtree)

#which hosts are in the parasite database
d<-read.csv("data/GMPD_main_2016-11-28.csv",header=T)
d<-subset(d,HostEnvironment=="terrestrial")
keep.d<-which(names(d)%in%c("HostCorrectedName","Host.Order","ParasiteCorrectedName","Par.Type","Group"))
d<-d[,keep.d]

#remove rows where host is not identified to species
d<-d[-which(d$HostCorrectedName=="no binomial name"),]

#remove rows where parasite is not identified to species
d<-d[-which(d$ParasiteCorrectedName=="no binomial name"),]
d<-d[-which(d$ParasiteCorrectedName=="not identified to genus"),]

#Sys.setlocale('LC_ALL','C') 
d<-d[-grep("sp.",d$ParasiteCorrectedName),]

## merge transmission mode data
ptrans<-read.csv("data/GMPD_parasite_traits_2016-11-28.csv",header=T)
ptrans<-subset(ptrans,select=c("ParasiteCorrectedName","close","nonclose","vector","intermediate"))
d<-merge(d,ptrans,by="ParasiteCorrectedName")

# identify host species in gmpd
obs.hosts<-unique(d$HostCorrectedName)
obs.hosts<-gsub(" ","_",obs.hosts) #put in underscore for compatibility with other dataframe

#at which locations of phydist do these hosts occur?
idx<-which(unlist(dimnames(phydist)[1]) %in% obs.hosts)

#pull out smaller cophenetic matrix of only these hosts
phydist.mini<-phydist[idx,idx]

#make a community matrix for the call to ses.mpd
d$HostCorrectedName<-gsub(" ","_",d$HostCorrectedName)
d$HostCorrectedName<-as.factor(d$HostCorrectedName)
d<-droplevels(d)
comm<-as.data.frame(table(d[,c(which(names(d)=="ParasiteCorrectedName"),which(names(d)=="HostCorrectedName"))]))
comm<-1*(comm>0)

#get normalized z-score for mpd

phydist.mini<-phydist.mini[order(rownames(phydist.mini)),order(rownames(phydist.mini))]

```

```{r, echo=F, eval=F}
nri<-ses.mpd(comm,phydist.mini,null.model="independentswap",runs=1000)#change back to 1000!
```

```{r, echo=F}
load("Rdata/get_nri.Rda")
```


```{r, echo=F, warning=F}
# prep to merge with other dataframe that has other parasite traits
#nri<-nri[complete.cases(nri),]  # THIS REMOVES SINGLETON PARASITES (ONLY ONE HOST) SINCE MPD STUFF IS NA
nri$ParasiteCorrectedName<-rownames(nri)
nri<-merge(nri,ptrans,by="ParasiteCorrectedName")

ptype<-read.csv("data/GMPD_parasite_taxonomy_2016-11-28.csv",header=T)
ptype<-subset(ptype,select=c("ParasiteCorrectedName","ParType"))
ptype<-ptype[!duplicated(ptype),]
nri<-merge(nri,ptype,by="ParasiteCorrectedName")

#remove rare parasite types
nri4<-nri[-which(nri$ParType %in% c("Fungus","Prion")),]#c("Fungus","Prion")

nri4$n.modes<-rowSums(cbind(nri4$close,nri4$nonclose,nri4$vector,nri4$intermediate))
nri6<-nri4
names(nri6)[which(names(nri6)=="ParasiteCorrectedName")]<-"para.name"
names(nri6)[which(names(nri6)=="ParType")]<-"para.type"

nri6$para.name<-as.character(nri6$para.name)

nri6<-nri6[complete.cases(nri6),]

nri.flat<-data.frame(para.name=character(),ntaxa=integer(),mpd.obs.z=numeric(),mpd.obs.p=numeric(),para.type=character(),tmode=character(),stringsAsFactors=FALSE)
k<-1
for (i in 1:dim(nri6)[1]){
  if (nri6[i,"close"]==1){nri.flat[k,"para.name"]<-nri6[i,"para.name"];nri.flat[k,"ntaxa"]<-nri6[i,"ntaxa"];nri.flat[k,"mpd.obs.z"]<-nri6[i,"mpd.obs.z"];nri.flat[k,"mpd.obs.p"]<-nri6[i,"mpd.obs.p"];nri.flat[k,"para.type"]<-as.character(nri6[i,"para.type"]);nri.flat[k,"tmode"]<-"close";k<-k+1}
  if (nri6[i,"nonclose"]==1){nri.flat[k,"para.name"]<-nri6[i,"para.name"];nri.flat[k,"ntaxa"]<-nri6[i,"ntaxa"];nri.flat[k,"mpd.obs.z"]<-nri6[i,"mpd.obs.z"];nri.flat[k,"mpd.obs.p"]<-nri6[i,"mpd.obs.p"];nri.flat[k,"para.type"]<-as.character(nri6[i,"para.type"]);nri.flat[k,"tmode"]<-"nonclose";k<-k+1}
  if (nri6[i,"vector"]==1){nri.flat[k,"para.name"]<-nri6[i,"para.name"];nri.flat[k,"ntaxa"]<-nri6[i,"ntaxa"];nri.flat[k,"mpd.obs.z"]<-nri6[i,"mpd.obs.z"];nri.flat[k,"mpd.obs.p"]<-nri6[i,"mpd.obs.p"];nri.flat[k,"para.type"]<-as.character(nri6[i,"para.type"]);nri.flat[k,"tmode"]<-"vector";k<-k+1}
  if (nri6[i,"intermediate"]==1){nri.flat[k,"para.name"]<-nri6[i,"para.name"];nri.flat[k,"ntaxa"]<-nri6[i,"ntaxa"];nri.flat[k,"mpd.obs.z"]<-nri6[i,"mpd.obs.z"];nri.flat[k,"mpd.obs.p"]<-nri6[i,"mpd.obs.p"];nri.flat[k,"para.type"]<-as.character(nri6[i,"para.type"]);nri.flat[k,"tmode"]<-"intermediate";k<-k+1}
}

```

Across all parasites there is a trend of infecting hosts that are more related than expected by chance, indicated by negative $z$-scores for mean pairwise PD between hosts (Fig. 1, global median $z$-score=-1.16). However, while 96% of parasites exhibit negative $z$-scores (trending to more specialist than expected by chance), only 15% of parasites have significantly negative $z$-scores. There are no parasites that have significantly positive $z$-scores.

```{r, echo=FALSE, warning=F, message=F}

nri.flat$para.type<-as.factor(nri.flat$para.type)
nri.flat$tmode<-as.factor(nri.flat$tmode)
nri.flat<-nri.flat[complete.cases(nri.flat),]
#order factors in z score order for plotting
ag.para.z<-aggregate(nri.flat$mpd.obs.z,by=list(nri.flat$para.type),FUN=mean)
ag.para.z<-ag.para.z[order(ag.para.z$x,decreasing=T),]
nri.flat$para.type<-factor(nri.flat$para.type,ag.para.z$Group.1)
ag.tmode.z<-aggregate(nri.flat$mpd.obs.z,by=list(nri.flat$tmode),FUN=mean)
ag.tmode.z<-ag.tmode.z[order(ag.tmode.z$x,decreasing=F),]
nri.flat$tmode<-factor(nri.flat$tmode,ag.tmode.z$Group.1)
gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}

global.median.z<-median(nri6$mpd.obs.z,na.rm=T)

ggplot(nri.flat, aes(x=para.type, y=mpd.obs.z, fill=tmode)) + geom_boxplot(notch=T) + theme(axis.text.x  = element_text(angle=90, vjust=0.5)) + scale_fill_discrete(name="Transmission\nmode",labels=c("Close","Complex","Vector","Envir"))+xlab("Parasite type")+ylab("Standard effect size of mean pairwise\nPD between hosts")+ylim(c(-5,1))+geom_hline(yintercept=global.median.z,colour="black")

#explanation for horn-shaped boxes: http://r.789695.n4.nabble.com/Strange-horns-on-notched-box-plots-td858004.html
```

Figure 1. Boxplot of $z$-scores in each parasite group stratified by transmission mode. Positive $z$-scores indicate that a parasite species tends to infect host species that are more distantly related than expected by chance (under the null model). Alternatively, negative $z$-scores indicate the opposite. In terms of deviation from the null model, 15% of $z$-scores are significantly negative (none of the positive $z$-scores are statistically significant). Solid horizontal line indicates the global median $z$-score.

 

The proportion of these significantly negative $z$-scores varies across parasite type, with smallest to largest proportions corresponding to bacteria, arthropods, helminths, protozoa and viruses (test for equality of proportions, $p$ < 0.001, Fig. 2 - top panel). The proportion of significantly negative $z$-scores is not significantly different across parasite transmission modes (test for equality of proportions, p=0.77, Fig. 2 - bottom panel).







```{r, echo=F, warning=F, message=F}
sig.z<-NULL #note all significant z scores are negative (checked independently)
for (i in unique(nri6$para.type)){
  tmp<-subset(nri6,para.type==i)
  sig.z<-rbind(sig.z,c(i,dim(tmp)[1],length(which(tmp$mpd.obs.p<0.05))))
}
sig.z<-as.data.frame(sig.z)
names(sig.z)<-c("para.type","n","x")
sig.z$n<-as.integer(as.character(sig.z$n))
sig.z$x<-as.integer(as.character(sig.z$x))
sig.z$y<-sig.z$n-sig.z$x

#prop.test(sig.z$x,sig.z$n)
sig<-NULL
for (i in 1:dim(sig.z)[1]){
  for(j in 1:sig.z[i,"x"]){
    sig<-rbind(sig,c(as.character(sig.z$para.type[i]),1))
  }
  for(j in 1:sig.z[i,"y"]){
    sig<-rbind(sig,c(as.character(sig.z$para.type[i]),0))
  }
}

sig<-as.data.frame(sig)
names(sig)<-c("para.type","sig")
sig$para.type<-factor(sig$para.type,levels=c("Bacteria","Arthropod","Helminth","Protozoa","Virus"))
sig.a<-ggplot(sig,aes(x=para.type,fill=sig))+
       geom_bar(position="fill")+
       scale_fill_discrete(name="Significant")+
       xlab("Parasite type")+
       ylab("Proportion of parasites significantly specialist")+
       scale_fill_manual(values=c("gray","dodgerblue4"))+
       coord_flip()

```



```{r, echo=F, warning=F, message=F}
sig.z<-NULL #note all significant z scores are negative (checked independently)
for (i in unique(nri.flat$tmode)){
  tmp<-subset(nri.flat,tmode==i)
  sig.z<-rbind(sig.z,c(i,dim(tmp)[1],length(which(tmp$mpd.obs.p<0.05))))
}
sig.z<-as.data.frame(sig.z)
names(sig.z)<-c("tmode","n","x")
sig.z$n<-as.integer(as.character(sig.z$n))
sig.z$x<-as.integer(as.character(sig.z$x))
sig.z$y<-sig.z$n-sig.z$x
#prop.test(sig.z$x,sig.z$n)
sig<-NULL
for (i in 1:dim(sig.z)[1]){
  for(j in 1:sig.z[i,"x"]){
    sig<-rbind(sig,c(as.character(sig.z$tmode[i]),1))
  }
  for(j in 1:sig.z[i,"y"]){
    sig<-rbind(sig,c(as.character(sig.z$tmode[i]),0))
  }
}
sig<-as.data.frame(sig)
names(sig)<-c("tmode","sig")
sig$tmode<-as.character(sig$tmode)
sig$tmode[which(sig$tmode=="close")]<-"Close"
sig$tmode[which(sig$tmode=="intermediate")]<-"Complex"
sig$tmode[which(sig$tmode=="nonclose")]<-"Envir."
sig$tmode[which(sig$tmode=="vector")]<-"Vector"

sig$tmode<-factor(sig$tmode,levels=c("Close","Complex","Envir.","Vector"))
sig.b<-ggplot(sig,aes(x=tmode,fill=sig))+geom_bar(position="fill")+xlab("Transmission mode")+ylab("")+scale_fill_discrete(name="Significant")+scale_fill_manual(values=c("gray","dodgerblue4"))+coord_flip()



grid.arrange(sig.a,sig.b, nrow = 2)
```

Figure 2. Proportion of parasites with host sets in which the hosts are more related than expected by chance, grouped by parasite type (top panel) and transmission mode (bottom panel)

However, across the full range of $z$-scores, each of the two covariates (parasite type and transmission mode) and their interaction explain differences between mean values (ANOVA: Parasite type - $p$ < 0.001, Transmission mode - $p$ = 0.003, Interaction - p<0.001, supplementary material).


Parasite type by transmission mode interactions are particularly driven by protozoa and viruses, and close contact and vector-borne transmission (Fig. 3); close contact transmission is associated with atypical extreme specialism in these groups and vector-borne transmission is associated with atypical generalism in these same groups. 



```{r, echo=F, warning=F, message=F, fig.width=10}

nri6$para.type<-droplevels(nri6$para.type)


fit <- lm(mpd.obs.z ~ para.type*close + para.type*vector + para.type*intermediate + para.type*nonclose, data = nri6)
#summary(fit)

p1<-sjp.grpfrq(nri6$mpd.obs.z,
           nri6$para.type,
           intr.var = nri6$close,
           axis.titles = c("",""),
           intr.var.labels = rep(c("0", "1"),5), 
           type = "boxplot",ylim=c(-6,2),
           title="Close",
           prnt.plot=F)

p2<-sjp.grpfrq(nri6$mpd.obs.z,
           nri6$para.type,
           intr.var = nri6$nonclose,
           axis.titles = c("",""),
           intr.var.labels = rep(c("0", "1"),5), 
           type = "boxplot",ylim=c(-6,2),
           title="Envir.",
           prnt.plot=F)

p3<-sjp.grpfrq(nri6$mpd.obs.z,
           nri6$para.type,
           intr.var = nri6$vector,
           axis.titles = c("",""),
           intr.var.labels = rep(c("0", "1"),5), 
           type = "boxplot",ylim=c(-6,2),
           title="Vector",
           prnt.plot=F)

p4<-sjp.grpfrq(nri6$mpd.obs.z,
           nri6$para.type,
           intr.var = nri6$intermediate,
           axis.titles = c("",""),
           intr.var.labels = rep(c("0", "1"),5), 
           type = "boxplot",ylim=c(-6,2),
           title="Complex",
           prnt.plot=F)


grid.arrange(p1$plot+scale_fill_manual(name="Parasite\ntype",labels=c("Arthropod","Bacteria","Helminth","Protozoa","Virus"),values=gg_color_hue(5)[c(2,3,1,5,4)]), p2$plot+scale_fill_manual(name="Parasite\ntype",labels=c("Arthropod","Bacteria","Helminth","Protozoa","Virus"),values=gg_color_hue(5)[c(2,3,1,5,4)]), p3$plot+scale_fill_manual(name="Parasite\ntype",labels=c("Arthropod","Bacteria","Helminth","Protozoa","Virus"),values=gg_color_hue(5)[c(2,3,1,5,4)]), p4$plot+scale_fill_manual(name="Parasite\ntype",labels=c("Arthropod","Bacteria","Helminth","Protozoa","Virus"),values=gg_color_hue(5)[c(2,3,1,5,4)]), nrow = 2,left=textGrob("Standard effect size of mean pairwise\nPD between hosts",rot=90))


```

Figure 3. Standard effect size of mean pairwise PD between hosts stratified by parasite type (colored bars) and presence (1) or absence (0) of each transmission mode (named subplots), indicating interactions between parasite type (particularly protozoa and viruses) and transmission mode (particularly close contact and vector-borne transmission)

<!-- If you are okay without it being a boxplot, it might be more intuitive to just have points with error bars, and these points could be slightly jittered to represent the difference in the presence/absence of transmission mode. Perhaps use open and closed circles (or some other difference in symbols) to indicate transmission mode as well? This is just an idea.   -->


In comparing the most abundant parasite group (helminths) with the group containing the greatest number of specialists (viruses) it is notable that viruses exhibit much wider variation in degree of generalism (Fig. 4), containing many extreme specialists (large, negative $z$-scores) but also many relative generalists (positive $z$-scores). By contrast, helminths are more consistent in their degree of generalism, and contain relatively few species that are at either extreme of the generalism-specialism continuum.



```{r, echo=FALSE, warning=F}
zdat<-nri6[which(nri6$para.type %in% c("Virus","Helminth")),]
zdat$para.type<-droplevels(zdat$para.type)
#zdat$para.type <- factor(zdat$para.type, c("Virus","Protozoa","Arthropod","Bacteria","Helminth"))
mV<-mean(zdat$mpd.obs.z[which(zdat$para.type=="Virus")],na.rm=T)
mH<-mean(zdat$mpd.obs.z[which(zdat$para.type=="Helminth")],na.rm=T)
ggplot(zdat)+geom_density(aes(x=mpd.obs.z,fill=para.type),alpha=0.3)+geom_vline(xintercept=mV,col=gg_color_hue(5)[4],lwd=2)+geom_vline(xintercept=mH,col=gg_color_hue(5)[1],lwd=2)+xlab("Standard effect size for mean pairwise PD")+ylab("Density")+scale_fill_manual(name="Parasite\ntype",values=gg_color_hue(5)[c(1,4)])+xlim(c(-6,2))
```

Figure 4. Probability density function of $z$-scores for mean pairwise PD between hosts of helminths (red) and viruses (blue). Helminths are more generalist on average (group mean values denoted by color-coded vertical lines), but the virus group contains more extreme parasites in terms of both specialism and generalism. 

The standard effect size of maximum PD in a parasite's host set provides a comparative measure of the span that a parasite exhibits across the host phylogenetic tree (Fig. 5). In addition, the standard effect size of the ratio of mean minimum to maximum PD measures the tendency for hosts of a parasite to aggregate in the mammal phylogeny. All parasite taxonomic groups exhibit variation in both these metrics. Protozoa, virus and helminth groups contain several parasite species whose span is significantly smaller than expected by chance (Fig. 5 - left of boxes in subplots, see supplementary material for histograms). Viruses and protozoa additionally contain some species whose hosts are more aggregated than expected by chance (Fig. 5 - below boxes in subplots, see supplementary material for histograms).  


<!-- I'm not really convinced that the distributions in figure 4 are any different. At least, I wouldn't hang my hat on that result. Seems like it could go into the supplement. -->



```{r, echo=F}
source("scripts/ses.maxD.R")
source("scripts/ses.mntd.to.maxD.R")
```

```{r, echo=F, eval=FALSE}
myMaxD<-ses.maxD(comm,phydist.mini,null.model="independentswap",runs=1000)
myRatio<-ses.mntd.to.maxD(comm,phydist.mini,null.model="independentswap",runs=1000)
#myMPD<-ses.mpd(comm,phydist.mini,null.model="independentswap",runs=5)
```

```{r, echo=F}
load("Rdata/get_myMaxD.Rda")
load("Rdata/get_myRatio.Rda")
```

```{r, echo=F, warning=F}
auxPtype<-read.csv("data/GMPD_parasite_taxonomy_2016-11-28.csv",header=T)
auxPtype<-subset(auxPtype,select=c("ParasiteCorrectedName","ParType"))
d<-merge(d,auxPtype,by="ParasiteCorrectedName")
P.type.binom<-d[,c("ParType","ParasiteCorrectedName")]
P.type.binom<-P.type.binom[!duplicated(P.type.binom),]
P.type.binom<-P.type.binom[order(P.type.binom$ParasiteCorrectedName),]
### UP TO HERE - RESOLVE MERGE ISSUE
myMaxD$ParasiteCorrectedName<-rownames(myMaxD)
myMaxD<-merge(myMaxD,ptype,by="ParasiteCorrectedName")
names(myMaxD)[which(names(myMaxD)=="ParType")]<-"para.type"
myMaxD$col.para<-"TBD"
for (i in 1:dim(myMaxD)[1]){
  if (myMaxD$para.type[i]=="Helminth"){myMaxD$col.para[i]<-"red"}#"orange"
  if (myMaxD$para.type[i]=="Bacteria"){myMaxD$col.para[i]<-"orange"}#"blue"
  if (myMaxD$para.type[i]=="Arthropod"){myMaxD$col.para[i]<-"gray20"}#"gray20"
  if (myMaxD$para.type[i]=="Virus"){myMaxD$col.para[i]<-"blue"}#"red"
  if (myMaxD$para.type[i]=="Fungus"){myMaxD$col.para[i]<-"chartreuse3"}#"green"
  if (myMaxD$para.type[i]=="Protozoa"){myMaxD$col.para[i]<-"magenta"}#"magenta"
  if (myMaxD$para.type[i]=="Prion"){myMaxD$col.para[i]<-"yellow"}#"yellow"
}

##### GGPLOT
gg.disp<-myMaxD
myRatio$ParasiteCorrectedName<-rownames(myRatio)
myRatio<-subset(myRatio,select=c("ParasiteCorrectedName","mntd.to.maxD.obs.z"))
gg.disp<-merge(gg.disp,myRatio,by="ParasiteCorrectedName")
names(gg.disp)[which(names(gg.disp)=="mntd.to.maxD.obs.z")]<-"mntd.z"
gg.disp<-subset(gg.disp,select=c(para.type,maxD.obs.z,mntd.z,col.para))
names(gg.disp)<-c("para.type","span.z","mntd2span.z","my.color")
gg.disp<-gg.disp[complete.cases(gg.disp),]
gg.disp<-gg.disp[which(gg.disp$span.z>-10),]
gg.disp<-gg.disp[-which(gg.disp$para.type%in%c("Fungus","Prion")),]
gg.disp$para.type<-droplevels(gg.disp$para.type)
circ_ecoli<-myRatio$mntd.to.maxD.obs.z[which(rownames(myRatio)=="Escherichia coli")]
circ_tcruzi<-myRatio$mntd.to.maxD.obs.z[which(rownames(myRatio)=="Trypanosoma cruzi")]
ann_text1<-data.frame(span.z=0.0,mntd2span.z=circ_ecoli,para.type=factor("Bacteria",levels=levels(gg.disp$para.type)),my.color="black")
ann_text2<-data.frame(span.z=0.0,mntd2span.z=circ_tcruzi,para.type=factor("Protozoa",levels=levels(gg.disp$para.type)),my.color="black")
ggplot(gg.disp,aes(span.z,mntd2span.z,colour=factor(my.color)))+geom_point()+facet_wrap(~para.type,nrow=2)+xlab("Standard effect size for maximum pairwise PD")+ylab("Standard effect size for ratio of mean minimum\npairwise PD to maximum pairwise PD")+theme(legend.position="none")+geom_segment(aes(x=-2,xend=2,y=-2,yend=-2),colour="gray50",size=0.2)+geom_segment(aes(x=2,xend=2,y=-2,yend=2),colour="gray50",size=0.2)+geom_segment(aes(x=-2,xend=2,y=2,yend=2),colour="gray50",size=0.2)+geom_segment(aes(x=-2,xend=-2,y=-2,yend=2),colour="gray50",size=0.2)+theme(panel.margin=grid::unit(0.75,"cm"))+geom_text(data=ann_text1,label="O",colour="black")+geom_text(data=ann_text2,label="O",colour="black")+scale_color_manual(values=gg_color_hue(5)[c(4,2,5,3,1)])
```

Figure 5. Scatter plot showing $x$: standard effect size of maximum pairwise PD of a parasite's host set and $y$: standard effect size of the ratio of mean minimum pairwise PD to maximum pariwise PD, colored by parasite type. The x-axis provides a comparative measure of a parasite's span across the host phylogenetic tree, and the y-axis provides a comparative measure of the aggregation of a parasite's hosts in the phylogenetic tree. Circled bacterial and protozoan parasites are _Escherichia coli_ and _Trypanosoma cruzi_, respectively. These parasites have a similar span ($x\approx0$), but very different host aggregation patterns (_E. coli_ randomly distributed host species, _T. cruzi_ clumpily distributed host species). Boxes encompassing $-2<x,y<2$ contain parasites whose $z$-scores are not statistically different from null model scores (random host sets).



```{r,echo=F, warning=F, fig.width=10}
#mammal supertree
treelist=read.nexus("data/WR_2005_ST.txt")
#best dates
mtree=treelist[[1]]
host.prune<-as.data.frame(rownames(phydist.mini))
names(host.prune)<-"host.spp"
host.prune$host.spp<-as.character(host.prune$host.spp)
host.prune$rand<-runif(dim(host.prune)[1],0,1)
rownames(host.prune)<-host.prune$host.spp
host.prune<-subset(host.prune,select="rand")
mtree2<-treedata(mtree,host.prune,warnings=F)
orig.tip.label<-mtree2[[1]]$tip.label
mtree2[[1]]$tip.label<-rep("",dim(host.prune)[1])
#mtree2[[1]]$tip.label<-rep("*",399)

#identify some para with max z score and smallest mean2max ratio
hosts.tcruzi<-unique(d$HostCorrectedName[which(d$ParasiteCorrectedName=="Trypanosoma cruzi")])
hosts.ecoli<-unique(d$HostCorrectedName[which(d$ParasiteCorrectedName=="Escherichia coli")])

my.colors<-rep("white",dim(host.prune)[1])
gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}
# COMPARE 2 PARASITES WITH BIG SPANS WHERE ONE IS CLUMPY AND ONE IS RANDOM, AND ONE IS REGULAR
idx1<-which(rownames(host.prune)%in%hosts.tcruzi)
idx2<-which(rownames(host.prune)%in%hosts.ecoli)
my.colors[idx1]<-gg_color_hue(5)[5]
my.colors[idx2]<-gg_color_hue(5)[3]
my.colors[intersect(idx1,idx2)]<-"black"

plot(mtree2[[1]],show.tip.label=T,edge.color="gray20",edge.width=0.5,direction="downwards",cex=1.2)
tiplabels(pch=22,col="gray80",bg=my.colors,cex=2.5,adj=c(0,-2.5))
```

Figure 6. Examples of clumpy vs. non-clumpy distributions of parasites of similar span in the host phylogenetic tree. The parasites are _Escherichia coli_ and _Tryapnosoma cruzi_, colored according to their groupings in Fig. 5.


The main analysis is centered on parasite species that infect at least two host species. Including parasite species for which only one terrestrial mammal host is identified in the database provides an opportunity to compare single-host parasites and multi-host parasites (Fig. 7). Quantitatively, single host parasites ($n$ = 653) are almost as common as multi-host parasites ($n$ = 749), and broadly reflect the composition of parasite taxonomy and transmission modes observed in multi-host parasites. Two notable exceptions are the virus group, which is under-represented among single-host parasites in general, and protozoan complex life-cycle parasites, which are more commonly single-host than multi-host. 


```{r, echo=F, warning=F}
#plot parasite data for single-host parasites
HP.table<-cbind(as.character(d$HostCorrectedName),as.character(d$ParasiteCorrectedName))
HP.table<-as.data.frame(HP.table)
HP.table<-HP.table[!duplicated(HP.table),]
names(HP.table)<-c("host.name","para.name")
para.nhosts<-table(HP.table$para.name)
#prep for ggplot
para.nhosts<-as.data.frame(para.nhosts)
names(para.nhosts)<-c("ParasiteCorrectedName","nhosts")
P.metrics<-subset(d,select=c("ParasiteCorrectedName","ParType","close","nonclose","vector","intermediate"))
P.metrics<-P.metrics[!duplicated(P.metrics),]
P.metrics<-merge(P.metrics,para.nhosts,by="ParasiteCorrectedName")
P.metrics<-P.metrics[complete.cases(P.metrics),]
P.metrics$multi.host<-ifelse(P.metrics$nhosts>1,1,0)
P.metrics.tmp<-subset(P.metrics,select=c("ParasiteCorrectedName","close","nonclose","vector","intermediate"))
P.metrics.melt<-melt(P.metrics.tmp,id.vars="ParasiteCorrectedName")
P.metrics.melt<-subset(P.metrics.melt,value==1)
P.metrics.tmp<-subset(P.metrics,select=c("ParasiteCorrectedName","ParType","multi.host"))
P.metrics.melt<-merge(P.metrics.melt,P.metrics.tmp,by="ParasiteCorrectedName")
P.metrics.melt<-subset(P.metrics.melt,ParType!="Prion")
P.metrics.melt$Par.Type<-droplevels(P.metrics.melt$ParType)
P.metrics.melt$multihost.status<-as.factor(ifelse(P.metrics.melt$multi.host==0,"Single-host","Multi-host"))
P.metrics.melt$multihost.status<-factor(P.metrics.melt$multihost.status,levels(P.metrics.melt$multihost.status)[2:1])
P.metrics.melt<-subset(P.metrics.melt,ParType!="Fungus")
P.metrics.melt<-droplevels(P.metrics.melt)
P.metrics.melt$Par.Type<-factor(P.metrics.melt$Par.Type,levels(P.metrics.melt$ParType)[c(2,1,3,5,4)])
P.metrics.melt$variable<-factor(P.metrics.melt$variable,levels(P.metrics.melt$variable)[c(1,4,3,2)])#2,1,4,3
ggplot(P.metrics.melt,aes(ParType,fill=variable))+geom_bar()+facet_wrap(~multihost.status)+xlab("Parasite type")+scale_fill_manual(values=gg_color_hue(4)[c(1,2,3,4)],name="Transmission\nmode",labels=c("Close","Complex","Vector","Envir."))+ylab("Count")+theme(axis.text.x=element_text(angle=90,hjust=1))
```

Figure 7. Number of single-host and multi-host (>1 host species) parasites as a function of parasite type and transmission mode. Parasites with more than one reported transmission mode (343/1402) are counted in each valid transmission mode category. 


One potential explanation for the number of parasite species associating with a host is the amount of time the host species has evolved since common ancestry. However, neither the host's evolutionary distinctiveness [@Redding2006] nor terminal branch length (Fig. 8 - top and bottom panels, respectively) are strongly associated with the number of known parasite species (and this is true when inspecting each host order separately - supplementary material). 

 

```{r, echo=F, warning=F}
### which hosts are the ones associated with 'single-host' parasites?
## this analysis is sensitive to observing rare events - a point worth making. 
H.metrics<-as.data.frame(table(d$HostCorrectedName))
names(H.metrics)<-c("host.name","n.records")
H.metrics$n.para<-0
for (i in 1:dim(H.metrics)[1]){
  p.set<-unique(d$ParasiteCorrectedName[which(d$HostCorrectedName==H.metrics$host.name[i])])
  H.metrics$n.para[i]<-length(p.set)
}

mtree2[[1]]$tip.label<-orig.tip.label
my.ed<-evol.distinct(mtree2[[1]],type="equal.splits")
names(my.ed)<-c("host.name","ed")
H.metrics<-merge(H.metrics,my.ed)
H.metrics$singleton<-ifelse(H.metrics$n.para==1,1,0)
H.order.binom.tmp<-subset(d,select=c("HostCorrectedName","Group"))
H.order.binom.tmp<-H.order.binom.tmp[!duplicated(H.order.binom.tmp),]
names(H.order.binom.tmp)<-c("host.name","order")
H.metrics<-merge(H.metrics,H.order.binom.tmp)
#use terminal branch length
terms<-mtree$edge[,2]<=Ntip(mtree)
terminal.edges<-as.data.frame(cbind(mtree$tip.label,mtree$edge.length[terms==T]))
names(terminal.edges)<-c("host.name","tbl")
terminal.edges$tbl<-as.numeric(as.character(terminal.edges$tbl))


H.metrics<-merge(H.metrics,terminal.edges,by="host.name")
ev.ed<-ggplot(H.metrics,aes(x=factor(n.para),y=ed))+geom_boxplot()+xlab("Number of parasite species per host species")+ylab("Host evolutionary\ndistinctiveness")+theme(axis.text.x = element_text(angle = 90, hjust = 1))
ev.bl<-ggplot(H.metrics,aes(x=factor(n.para),y=tbl))+geom_boxplot()+xlab("Number of parasite species per host species")+ylab("Host terminal\nbranch length")+theme(axis.text.x = element_text(angle = 90, hjust = 1))

grid.arrange(ev.ed,ev.bl,ncol=1)
```

Figure 8. Box-and-whisker plots showing the relationship between number of parasite species per host species and host evolutionary distinctiveness (top panel) and host terminal branch length (bottom panel).

<!-- What is the null expectation of the decay of host evolutionary distinctiveness with parasite species richness (top panel)? Is the argument that more evolutionary removed host species tend to have fewer parasite species?  -->


The number of parasite species per host species is over dispersed (variance to mean ratio ~25), meaning that a small number of host species are associated with a large number of parasite species. Accordingly, a negative binomial generalized linear model was used to predict the number of parasite species by each measure of evolutionary distinctivess (and, optionally, the number of host records in GMPD, which was used to control for sampling bias). Only the model with evolutionary distinctiveness (but not the number of host recrods) showed a significant relationship (supplementary material). 








## Discussion

We have shown that most multi-host parasites are relatively generalist. However, among the ~750 multi-host parasite species considered here, some infect host species that are comparatively closely related. Of the five parasite groups, bacteria are the most generalist on average and protozoan and viral parasite groups contain the most specialist parasites. While transmission mode naturally impacts the opportunity for parasites to encounter novel host species, it was less influential in determining parasite generalism than parasite taxonomy. However, there were interactions between parasite type and transmission mode; both protozoan and viral parasites exhibit specialism when undergoing close contact transmission, in agreement with primate parasite research [@Pedersen2005]. Additionally, protozoan parasites that are environmentally transmitted are often more specialist than expected by chance, whereas vector-borne protozoa are typically generalist. 

Previous research based on taxonomic definitions of generalism have tended to suggest that viruses and protozoa are relatively generalist and helminths relatively specialist [@Pedersen2005]. The differences between those results and the ones we present here lie in the definition of generalism, which can exagerate rare but large host species jumps. We found, for example, that while viruses often infect closely related hosts (shown by large negative $z$-scores), they are also a group that contains several parasites whose hosts are distantly related. In concert with purely taxonomic definitions of generalism, such as the ability to infect hosts of different orders, examining the standard effect sizes of mean hostPD, span and clustering complements this approach with continuous, standardized measures that facilitate comparison across parasite species and with null models.  

While the underlying data represent known host-parasite occurrences, they are not necessarily indicative of parasite fitness. Parasites may jump between host species, establishing in each, e.g. Rabies virus [@Holmes2009], or maintain themselves in some reservoir host species and spill over to others, e.g. West Nile virus [@Reisen2013]. Consequently, it is difficult to infer future potential for novel host acquisition from existing data. Plausibly, parasites with rapid evolution may be both good adaptors to, and explorers of, the space of host species, as is indicated by the virus group which exhibits clustering of host species in the mammal phylogeny, which is indicative of taking occasional leaps to novel host species and subsequently colonizing closely related host species.

Host species geographical ranges have previously been shown to be one of the strongest predictors of viral parasite sharing among primates [@Davies2008], and are only implicitly included here due to the non-independence of range overlap and phylogenetic relatedness [@Poulin2008]. Explicit inclusion of geography is a promising line of macroecological inquiry.










## Supplementary material


### Parasite taxonomy, transmission mode and their interaction contribute to variation in mean pairwise PD of host sets of each parasite

```{r, echo=T}
#simple test to motivate investigation of factors and their interaction
basic.aov<-aov(lm(nri.flat$mpd.obs.z~nri.flat$para.type*nri.flat$tmode))
summary(basic.aov)
```

### Parasite transmission mode as a predictor of span and aggregation of hosts in the mammalian phylogeny

```{r, echo=F, warning=F, message=F}
gg.disp$ParasiteCorrectedName<-"tbd"
for (i in 1:dim(gg.disp)[1]){
  idx<-as.integer(rownames(gg.disp)[i])
  gg.disp$ParasiteCorrectedName[i]<-myMaxD$ParasiteCorrectedName[idx]
}
gg.disp<-merge(gg.disp,ptrans,by="ParasiteCorrectedName")
gg.disp$tmode<-"mixed"
for (i in 1:dim(gg.disp)[1]){
  if (gg.disp$close[i]==1&gg.disp$nonclose[i]==0&gg.disp$vector[i]==0&gg.disp$intermediate[i]==0){gg.disp$tmode[i]<-"close"}
  if (gg.disp$close[i]==0&gg.disp$nonclose[i]==1&gg.disp$vector[i]==0&gg.disp$intermediate[i]==0){gg.disp$tmode[i]<-"nonclose"}
  if (gg.disp$close[i]==0&gg.disp$nonclose[i]==0&gg.disp$vector[i]==1&gg.disp$intermediate[i]==0){gg.disp$tmode[i]<-"vector"}
  if (gg.disp$close[i]==0&gg.disp$nonclose[i]==0&gg.disp$vector[i]==0&gg.disp$intermediate[i]==1){gg.disp$tmode[i]<-"intermediate"}
}
ggplot(gg.disp,aes(span.z,mntd2span.z,colour=my.color))+geom_point()+facet_wrap(~tmode,nrow=2)+xlab("Standard effect size for maximum pairwise PD")+ylab("Standard effect size for ratio of mean minimum\npairwise PD to maximum pairwise PD")+theme(legend.position="none")+geom_segment(aes(x=-2,xend=2,y=-2,yend=-2),colour="gray50",size=0.2)+geom_segment(aes(x=2,xend=2,y=-2,yend=2),colour="gray50",size=0.2)+geom_segment(aes(x=-2,xend=2,y=2,yend=2),colour="gray50",size=0.2)+geom_segment(aes(x=-2,xend=-2,y=-2,yend=2),colour="gray50",size=0.2)+theme(panel.margin=grid::unit(0.5,"cm"))+scale_color_manual(values=gg_color_hue(5)[c(4,2,5,3,1)])
```

Each transmission mode group contains a majority of parasites species whose standard effect sizes for span and aggregation are not different those of equivalent random host sets. All groups contain some parasites that have a smaller span than expected by change and contain some parasites that exhibit patterns of host species phylogenetic aggregation. Patterns are relatively similar across all groups. 

### Host span and clutering tendency as function of transmission mode

```{r, echo=F, warning=F, message=F}
ggplot(gg.disp,aes(x=span.z))+geom_histogram()+facet_wrap(~para.type)+geom_vline(xintercept=-1.96,color="red")+xlab("Host span (standard effect size)")
```

Components of distributions left of vertical red lines have smaller spans than expected by chance.

```{r, echo=F, warning=F, message=F}
ggplot(gg.disp,aes(x=mntd2span.z))+geom_histogram()+facet_wrap(~para.type)+geom_vline(xintercept=-1.96,color="red")+xlab("Host clustering (standard effect size)")

```

Components of distributions left of vertical red lines exhibit greater host clustering than expected by chance.

### Effects of host orders on relatedness of a parasite's host set

Several parasites exclusively infect hosts within a host order. Grouping host species by order allows us to inspect how results on generalism are driven by the non-monophyletic structure of a tree with three host orders. Here, the null models are constructed in a similar way to the main text, except that random mammalian host species are selected according to the host taxonomic bias of the parasite. For example, a parasite known to infect 5 carnivores, 5 primates and 0 ungulates would have its random sets constrcuted from 5 randomly selected carnivores and 5 randomly selected primates. The plot separates parasites that infect hosts of multiple orders, carnivores only, primates only and ungulates only. 

```{r, echo=FALSE, eval=F, warning=F}
nri6$z.tax<--999
##calc. n hosts of each order for each parasite
nri6$n.carnivores<-0
nri6$n.primates<-0
nri6$n.ungulates<-0
for (i in 1:dim(nri6)[1]){
  this.para<-nri6$para.name[i]
  host.set<-unique(d$HostCorrectedName[which(d$ParasiteCorrectedName==this.para)])
  nri6$n.carnivores[i]<-sum(H.metrics$order[H.metrics$host.name %in% host.set]=="carnivores")
  nri6$n.primates[i]<-sum(H.metrics$order[H.metrics$host.name %in% host.set]=="primates")
  nri6$n.ungulates[i]<-sum(H.metrics$order[H.metrics$host.name %in% host.set]=="ungulates")
}

H.metrics$host.name<-as.character(H.metrics$host.name)
for (i in 1:dim(nri6)[1]){
  the.mpds<-NULL
  n.reps<-1000
  for (rep in 1:n.reps){# need to do this several times to build up distrib. (n=1000, 10 for testing)
    this.set<-c(sample(H.metrics$host.name[H.metrics$order=="carnivores"],size=nri6$n.carnivores[i],replace=F),sample(H.metrics$host.name[H.metrics$order=="primates"],size=nri6$n.primates[i],replace=F),sample(H.metrics$host.name[H.metrics$order=="ungulates"],size=nri6$n.ungulates[i],replace=F))
    sub.phy<-phydist.mini[which(dimnames(phydist.mini)[[1]] %in% this.set),which(dimnames(phydist.mini)[[1]] %in% this.set)]
    the.mpds<-c(the.mpds,mean(sub.phy[lower.tri(sub.phy==T,diag=F)]))
  }
  nri6$z.tax[i]<-(nri6$mpd.obs[i]-mean(the.mpds))/sd(the.mpds)
}

```

```{r, echo=F, eval=T}
load("Rdata/get_nri6ztax.Rda")
```

```{r, echo=F}
nri6$host.orders<-"TBA"
for (i in 1:dim(nri6)[1]){
  if (nri6$n.carnivores[i]>0&nri6$n.primates[i]==0&nri6$n.ungulates[i]==0){nri6$host.orders[i]<-"Carnivores"}
  else if (nri6$n.carnivores[i]==0&nri6$n.primates[i]>0&nri6$n.ungulates[i]==0){nri6$host.orders[i]<-"Primates"}
  else if (nri6$n.carnivores[i]==0&nri6$n.primates[i]==0&nri6$n.ungulates[i]>0){nri6$host.orders[i]<-"Ungulates"}
  else {nri6$host.orders[i]<-"Multi-order"}
}
ggplot(nri6,aes(x=para.type,y=z.tax,fill=para.type))+geom_boxplot()+theme(axis.text.x=element_text(angle=90,vjust=0.5))+xlab("Parasite type")+ylab("Standard effect size of mean pairwise phylogenetic distance between hosts")+facet_wrap(~host.orders)+scale_fill_manual(name="Parasite\ntype",labels=c("Arthropod","Bacteria","Helminth","Protozoa","Virus"),values=gg_color_hue(5)[c(2,3,1,5,4)])
```

The patterns for standard effect size of mean pairwise PD of all host species is broadly reflected in the patterns of specific host orders. For viruses infecting only one host order, the carnivore host group is particularly associated with specialist viruses.

```{r, echo=F}
nri7<-nri6[which(nri6$n.modes==1),]
nri7$tmode<-"TBA"
for (i in 1:dim(nri7)[1]){
  if (nri7$close[i]==1){nri7$tmode[i]<-"Close"}
  if (nri7$nonclose[i]==1){nri7$tmode[i]<-"Envir"}
  if (nri7$intermediate[i]==1){nri7$tmode[i]<-"Complex"}
  if (nri7$vector[i]==1){nri7$tmode[i]<-"Vector"}
}
ggplot(nri7,aes(x=tmode,y=z.tax,fill=tmode))+geom_boxplot()+theme(axis.text.x=element_text(angle=90,vjust=0.5))+xlab("Transmission mode")+ylab("Standard effect size of mean pairwise phylogenetic distance between hosts")+facet_wrap(~host.orders)+scale_fill_manual(name="Transmission\nmode",values=gg_color_hue(4)[c(1,2,3,4)])
```

Parasites affecting only primates and ungulates do not exhibit differences in specialism/generalism due to parasite transmission mode. The carnivore-exclusive parasites show a similar trend to the main result (Figs. 1 & 2). 


### Negative binomial modeling of parasite species richness 

Summary of negative binomial GLMs with predictor variables selected from: evolutionary distinctiveness (ed), terminal branch length (tbl) and number of host records in GMPD (n.records), and response variable number of parasite species per host species (n.para)


```{r, echo=F, eval=T}
VMR<-var(H.metrics$n.para)/mean(H.metrics$n.para)
m.ed<-glm.nb(n.para~ed,data=H.metrics,link=log,maxit=100)
m.bl<-glm.nb(n.para~tbl,data=H.metrics,link=log,maxit=100)
m.ed2<-glm.nb(n.para~ed+n.records,data=H.metrics,link=log,maxit=100)
m.bl2<-glm.nb(n.para~tbl+n.records,data=H.metrics,link=log,maxit=100)

```

```{r, results='as-is', echo=FALSE}

xtable(m.ed)
xtable(m.bl)
xtable(m.ed2)
xtable(m.bl2)

```


For models using host evolutionary distinctiveness or terminal branch length (with and without number of host records, to control for potential sampling bias), to predict the number of parasite species per host species, the main predictor variable is only significant for evolutionary disctinctiveness without controlling for number of host records. 


### Lack of association between host evolutionary distinctiveness and number of parasite species is exhibited across host orders.

```{r, echo=F}
ho.ed<-ggplot(H.metrics,aes(x=as.factor(n.para),y=ed))+geom_boxplot()+facet_wrap(~order)+theme(axis.text.x=element_text(angle=90,hjust=1))+scale_x_discrete(breaks=seq(0,130,10))+xlab("Number of parasite species per host species")+ylab("Host evolutionary\ndistinctiveness")
ho.bl<-ggplot(H.metrics,aes(x=as.factor(n.para),y=tbl))+geom_boxplot()+facet_wrap(~order)+theme(axis.text.x=element_text(angle=90,hjust=1))+scale_x_discrete(breaks=seq(0,130,10))+xlab("Number of parasite species per host species")+ylab("Host terminal\nbranch length")
grid.arrange(ho.ed,ho.bl,ncol=1)
```

When separately inspecting the relationship between evolutionary distinctiveness and number of parasite species per host species within host orders (carnivores, primates, ungulates) there is no significant relationship in any host order. 

## References
